src/sentry/dynamic_sampling/rules/biases/custom_rule_bias.py

```diff
@@ -0,0 +1,94 @@
+"""
+Bias for custom rules (i.e. user-defined rules).
+"""
+import hashlib
+from datetime import datetime
+from typing import Any, List, Mapping, TypedDict
+
+from sentry.dynamic_sampling import get_redis_client_for_ds
+from sentry.snuba.metrics.extraction import RuleCondition
+from sentry.utils import json
+
+
+class SerializedRule(TypedDict, total=True):
+    """
+    The format of the rule data stored in Redis
+
+    All rules for one org are stored in a Redis hash.
+    NOTE: the key only contains the condition you can't have the same condition
+    applied to different projects with different expiration dates.
+    If a condition should be applied to many projects a single rule should be created
+
+    """
+
+    condition: RuleCondition  # the rule condition parsed from the query
+    expiration: float  # the timestamp of the rule's expiration (UNIX timestamp)
+    project_ids: List[int]  # list of project ids applicable to this rule
+    org_id: int  # the organization id
+
+
+def get_rule_hash(rule: SerializedRule) -> str:
+    """
+    Returns the hash of the rule based on the condition
+    """
+    condition_hash = hash_value(rule["condition"])
+    # make it a bit shorter
+    condition_hash = hashlib.sha1(condition_hash.encode("utf-8")).hexdigest()
+    return condition_hash
+
+
+def rule_from_json(val: str) -> SerializedRule:
+    """
+    Creates a SerializedRule from a json string
+    """
+    return json.loads(val)
+
+
+def custom_rules_redis_key(org_id: int) -> str:
+    """
+    Returns the Redis key for the custom rules for an organization
+    """
+    return f"ds::o:{org_id}:custom-rules"
+
+
+def hash_value(val: Any) -> str:
```

Comment:
This function name is a bit ambiguous since it can imply a hashing action or that the value is needed for the hash.

I would suggest a name that connects better with what the function does, regardless of what callers do with it:
```suggestion
def to_sorted_string(val: Any) -> str:
```